File: _archive/nakadi/java-nakadi-demo/src/main/java/de/felixroske/javanakadidemo/HelloworldNakadiListener.java
package de.felixroske.javanakadidemo;

import org.springframework.stereotype.Component;
import org.zalando.spring.boot.fahrschein.nakadi.NakadiListener;

import java.util.List;
import java.util.Locale;

@Component
public class HelloworldNakadiListener implements NakadiListener<HelloWorld> {
    @Override
    public Class<HelloWorld> getEventType() {
        return HelloWorld.class;
    }

    @Override
    public void accept(List<HelloWorld> events) {
        for (HelloWorld o : events) {
            System.out.println(o.getName().toUpperCase(Locale.GERMAN));
        }
    }
}

File: _archive/nakadi/kotlin-nakadi-demo/src/main/kotlin/de/felixroske/nakadidemo/HelloworldNakadiListener.kt
package de.felixroske.nakadidemo

import org.springframework.stereotype.Component
import org.zalando.spring.boot.fahrschein.nakadi.NakadiListener

class Helloworld {
    var name: String = ""
    var age: Int = 0
}

@Component
class HelloworldNakadiListener: NakadiListener<Helloworld> {

    override fun accept(events: MutableList<Helloworld>) {
        for (event in events) {
            println("${event.name} - ${event.age}")
        }
    }

    override fun getEventType(): Class<Helloworld> {
        return Helloworld::class.java
    }
}